### PR TITLE
app-crypt/swtpm: Fix conditional patching

### PR DESCRIPTION
**Pull request by salahcoronya**
While not absolutely verboten, conditional patching is discouraged in Gentoo. This PR removed the conditional patches, and provides a "proper" patch for configure.ac do the condition is no longer needed. No functional changes.

### CODE REVIEW

File: app-crypt/swtpm/files/swtpm-0.7.2-Conditionalize-test-dependencies.patch [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+Date: Mon, 28 Mar 2022 22:10:07 -0500
+Subject: [PATCH] configure.ac: Conditionalize test dependencies
+
+Signed-off-by: Christopher Byrne <salah.coronya@gmail.com>
+---
+ configure.ac | 25 +++++++++++++++++--------
+ 1 file changed, 17 insertions(+), 8 deletions(-)
+
+diff --git a/configure.ac b/configure.ac
+index e9f9d8f..c7dc3bf 100644
```
Comment (laumann) [inline app-crypt/swtpm/files/swtpm-0.7.2-Conditionalize-test-dependencies.patch:12]: Awesome! Thanks for making this change. Could I ask you scrub the patch as outlined in the [Clean patch howto](https://devmanual.gentoo.org/ebuild-writing/misc-files/patches/index.html#clean-patch-howto)?

File: app-crypt/swtpm/files/swtpm-0.7.2-Conditionalize-test-dependencies.patch [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+-	AC_MSG_ERROR([expect is required: expect package])
+-fi
++AC_ARG_ENABLE([test],
++	[AS_HELP_STRING([--enable-test],
++			[enable tests (default is yes)])],
++	[enable_test=$enableval],
++	[enable_test=yes])
++
++AS_IF([test "x$enable_test" != xno],
++	[AC_PATH_PROG([EXPECT], expect)
```
[the remark this answers is not in the document]
Comment (salahcoronya) [reply app-crypt/swtpm/files/swtpm-0.7.2-Conditionalize-test-dependencies.patch:28]: Ok changed the if's to AS_IF in my changes,

File: app-crypt/swtpm/files/swtpm-0.7.2-Conditionalize-test-dependencies.patch (hunk 1 above)
Comment (laumann) [reply app-crypt/swtpm/files/swtpm-0.7.2-Conditionalize-test-dependencies.patch]: OK, the headers generated by `git send-email` aren't necessary either though nor is the diffstat. Basically everything from `--- a/configure.ac` and up can be removed.

Comment (salahcoronya) [reply app-crypt/swtpm/files/swtpm-0.7.2-Conditionalize-test-dependencies.patch]: OK, all removed

Comment (thesamesam) [reply app-crypt/swtpm/files/swtpm-0.7.2-Conditionalize-test-dependencies.patch]: I slightly disagree in that authorship is useful as is the subject, but yes, scrubbing is good.

Comment (laumann) [reply app-crypt/swtpm/files/swtpm-0.7.2-Conditionalize-test-dependencies.patch]: I agree, I wasn't very precise here. Including a brief [patch description](https://devmanual.gentoo.org/ebuild-writing/misc-files/patches/index.html#patch-descriptions) above the diff is encouraged :-)

**Comment by salahcoronya** [comment]
Ping?

**Review by juippis — APPROVED** [review]
Thanks!

**Pull request closed without merging**